sim/team-validator.ts

```diff
@@ -1767,6 +1775,13 @@ export class TeamValidator {
 		}
 		// Event-related ability restrictions only matter if we care about illegal abilities
```

Comment:
This comment is a bit confusing now that you've added a check for illegal moves.

Reply:
is this comment even necessary?

Reply:
Yeah, you can take it out.